packages/image_picker/image_picker/ios/Classes/FLTImagePickerPlugin.m

```diff
@@ -246,12 +321,86 @@ - (void)errorNoPhotoAccess:(PHAuthorizationStatus)status {
   }
 }
 
-- (void)showPhotoLibrary {
+- (void)showPhotoLibrary:(ImagePickerClassType)imagePickerClassType {
   // No need to check if SourceType is available. It always is.
-  _imagePickerController.sourceType = UIImagePickerControllerSourceTypePhotoLibrary;
-  [[self viewControllerWithWindow:nil] presentViewController:_imagePickerController
-                                                    animated:YES
-                                                  completion:nil];
+  switch (imagePickerClassType) {
+    case PHPickerClassType:
+      [[self viewControllerWithWindow:nil] presentViewController:_pickerViewController
+                                                        animated:YES
+                                                      completion:nil];
+      break;
+    case UIImagePickerClassType:
+      _imagePickerController.sourceType = UIImagePickerControllerSourceTypePhotoLibrary;
+      [[self viewControllerWithWindow:nil] presentViewController:_imagePickerController
+                                                        animated:YES
+                                                      completion:nil];
+      break;
+  }
+}
+
+- (NSNumber *)getDesiredImageQuality:(NSNumber *)imageQuality {
+  if (![imageQuality isKindOfClass:[NSNumber class]]) {
+    imageQuality = @1;
+  } else if (imageQuality.intValue < 0 || imageQuality.intValue > 100) {
+    imageQuality = [NSNumber numberWithInt:1];
+  } else {
+    imageQuality = @([imageQuality floatValue] / 100);
+  }
+  return imageQuality;
+}
+
+- (void)picker:(PHPickerViewController *)picker
+    didFinishPicking:(NSArray<PHPickerResult *> *)results API_AVAILABLE(ios(14)) {
+  [picker dismissViewControllerAnimated:YES completion:nil];
+
+  NSNumber *maxWidth = [_arguments objectForKey:@"maxWidth"];
+  NSNumber *maxHeight = [_arguments objectForKey:@"maxHeight"];
+  NSNumber *imageQuality = [_arguments objectForKey:@"imageQuality"];
+  NSNumber *desiredImageQuality = [self getDesiredImageQuality:imageQuality];
+
+  for (PHPickerResult *result in results) {
+    [result.itemProvider
+        loadObjectOfClass:[UIImage class]
+        completionHandler:^(__kindof id<NSItemProviderReading> _Nullable image,
+                            NSError *_Nullable error) {
+          if ([image isKindOfClass:[UIImage class]]) {
+            if (image != nil) {
```

Comment:
Shouldn't the `nil` check occur before the `isKindOfClass `?

Reply:
Good catch, we actually don't need this nil check since `[image isKindOfClass:[UIImage class]]` returns false if image is nil.